Open Judge System/Workers/OJS.Workers.ExecutionStrategies/RubyExecutionStrategy.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System.IO;
+
+    using OJS.Common.Extensions;
+    using OJS.Workers.Checkers;
+    using OJS.Workers.Common;
+    using OJS.Workers.Executors;
+
+    public class RubyExecutionStrategy : ExecutionStrategy
+    {
+
+        public RubyExecutionStrategy(string rubyPath)
+        {
+            this.RubyPath = rubyPath;
+        }
+   
+        public string RubyPath { get; set; }
+
+        public string WorkingDirectory { get; set; }
+
+        public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            var result = new ExecutionResult();
+
+            result.IsCompiledSuccessfully = true;
+
+            string submissionFilePath = FileHelpers.SaveStringToTempFile(executionContext.Code);
+
+            var arguments = new[] { submissionFilePath };
+
+            IExecutor executor = new RestrictedProcessExecutor();
+            var checker = Checker.CreateChecker(
+                executionContext.CheckerAssemblyName,
+                executionContext.CheckerTypeName,
+                executionContext.CheckerParameter);
+
+            foreach (var test in executionContext.Tests)
+            {
+                var processExecutionResult = executor.Execute(
+                    this.RubyPath,
+                    test.Input,
+                    executionContext.TimeLimit,
+                    executionContext.MemoryLimit,
+                    arguments,
+                    this.WorkingDirectory);
```

Comment:
You can use the overload that doesn't require working directory, when you remove it.